game/src/info/mod.rs

```diff
@@ -528,14 +528,15 @@ impl InfoPanel {
                     (
                         false,
                         Some(Transition::ReplaceWithData(Box::new(
-                            move |state, ctx, app| {
-                                let mut sandbox = state.downcast::<SandboxMode>().ok().unwrap();
-
+                            move |mut state, ctx, app| {
                                 if time < app.primary.sim.time() {
-                                    sandbox = ctx.loading_screen("rewind simulation", |ctx, _| {
-                                        Box::new(SandboxMode::new(ctx, app, sandbox.gameplay_mode))
+                                    let gameplay =
+                                        state.downcast::<SandboxMode>().ok().unwrap().gameplay_mode;
+                                    state = ctx.loading_screen("rewind simulation", |ctx, _| {
```

Comment:
I'm having a little trouble understanding the significance of this change - why are we mutating state in this new way, and getting gameplay differently?

Reply:
This was a little unrelated to this PR. It's a more ergonomic API for state constructors to return `Box<dyn State>`. But originally, I was making the caller do the boxing. I happened to clean this up for `SandboxMode` here, and this is the one weird consequence. When we rewind time, it's cleanest to just create a SandboxMode from scratch. The alternative to the rewrite here would be something like `sandbox = SandboxMode::new(ctx, app, sandbox.gameplay_mode).downcast::<SandboxMode>().ok().unwrap()`. That feels a little more confusing than this way